### PR TITLE
Include ha settings in instance settings endpoint

### PR DESCRIPTION
**Pull request by hardillb**
## Description

Disable editor if HA enabled.

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: forge/routes/api/project.js [hunk 1]
```diff
@@ -786,6 +786,15 @@ module.exports = async function (app) {
...
             delete settings.settings.env
         }
+
+        const ha = await request.project.getSetting('ha')
+        if (ha && ha.replicas > 1) {
+            settings.ha = ha
+            // not sure about this, is it something the launcher should do based on settings.ha being present?
+            // also doesn't disable the link in the FlowForge app
+            settings.settings.disableEditor = true
+        }
```
Comment (knolleary) [inline forge/routes/api/project.js:796]:
```suggestion
        if (app.config.features.enabled('ha')) {
            const ha = await request.project.getHASettings()
            if (ha && ha.replicas > 1) {
                settings.ha = ha
            }
        }
```
We should push disabling the editor further down the stack. In the future we may be able to enable the editor in HA mode, but will depend on the NR version (I have the start of an idea for NR 4.x) - so we should push the decision further down.

**Review by knolleary — CHANGES_REQUESTED** [review]

**Comment by knolleary** [comment]
@hardillb also needs a test adding

**Comment by hardillb** [comment]
OK, messed something up merging changes, will fix

File: test/unit/forge/routes/api/project_spec.js [hunk 2]
```diff
@@ -851,6 +851,24 @@ describe('Project API', function () {
                 result.should.have.property('ha')
                 result.ha.should.have.property('replicas', 2)
             })
+
+            it('Check Project Settings have ha key', async function () {
+                const project = await createInstance()
+                project.updateHASettings({
```
Comment (knolleary) [inline test/unit/forge/routes/api/project_spec.js:857]:
```suggestion
                await project.updateHASettings({
```

**Review by knolleary — APPROVED** [review]

**Pull request merged**